Makefile

```diff
@@ -62,7 +62,7 @@ travis_push::
 	$(MAKE) banner
 	$(MAKE) ensure
 ifeq ($(TRAVIS_BRANCH),master)
-	HUGO_ENVIRONMENT=production $(MAKE) build
```

Comment:
There are places where we make use of `{{- if ne hugo.Environment "production" }}` and `{{- if eq hugo.Environment "production" }}` checks. Do those still work now that we're no longer setting `HUGO_ENVIRONMENT=production` explicitly? Or does running `hugo` use `production` by default (whereas `hugo server` uses `development` by default)? (I can't recall off the top of my head.)